sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/Exchange.scala

```diff
@@ -52,6 +54,13 @@ abstract class Exchange extends UnaryExecNode {
 case class ReusedExchangeExec(override val output: Seq[Attribute], child: Exchange)
   extends LeafExecNode {
 
+  override def equals(that: Any): Boolean = that match {
```

Comment:
A `ReusedExchangeExec` is a reference to an `Exchange`, so 2 `ReusedExchangeExec`s should be equal only if they are pointing to the same exchange instance.